src/toolbar/operatortoolbar.tsx

```diff
@@ -111,7 +126,14 @@ export class OperatorToolbarReact extends React.Component<IProps> {
           const model = this.props.toolbarModel.sharedModel;
           if (model) {
             model.transact(() => {
-              model.addObject(objectModel);
+              if (!model.objectExists(objectModel.name)) {
+                model.addObject(objectModel);
```

Comment:
Should we make the `objectExists` check under `addObject` instead? So we don't need to add that check code everywhere we use `addObject`?

Reply:
`addObject` also checks whether it exists or not. The reason I added `objectExists` is to be able to check before trying to add the element and show an error mesage if it exists.